salt/proxy/junos.py

```diff
@@ -150,6 +160,15 @@ def ping():
     '''
     Ping?  Pong!
     '''
+
+    try:
+        thisproxy['conn'].rpc('<file-list><path>/dev/null</path></file-list>')
+
+    except RpcTimeoutError:
+        thisproxy['conn'].connected = False
+    except ConnectClosedError:
+        thisproxy['conn'].connected = False
+
     return thisproxy['conn'].connected
```

Comment:
Rather than duplicating the code of an entire function, could you just have `ping()` invoke `alive()`?